op-e2e/sequencer_failover_setup.go

```diff
@@ -299,26 +299,46 @@ func sequencerCfg(rpcPort int) *rollupNode.Config {
 	}
 }
 
-func waitForLeadershipChange(t *testing.T, c *conductor, leader bool) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+func waitFor(t *testing.T, duration time.Duration, condition func() bool) error {
```

Comment:
I think this could just be replaced by the utilities in the `wait` package like `wait.For`

Reply:
Addressed in my new PR: https://github.com/ethereum-optimism/optimism/pull/9576